File: greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float m, a, b;
int q, d, n, p;

int main(void)

{
     do
    {
        printf("Change owed = ");
        m = get_float();
    }
    while
        (m <= 0);

m = m * 100;
m = (round) (m);

{printf("m = %f\n", m);}
while (m >= 25)
{
    m = m - 25;
    q++;
}
{printf("q = %i\n", q);}
a = (int) (m) % 25;

while (a >= 10)
{
	a = a - 10;
	d++;
}
{printf("d = %i\n", d);}
b = (int) (a) % 10;

while (b >= 5)
{
	b = b - 5;
	n++;
}
{printf("n = %i\n", n);}

p = (int) (b) % 5;

{printf("p = %i\n", p);}
{
    printf("Minimum number of coins used is %i\n", q + d + n + p);
}

}

File: greedy.py
import cs50

def main():
    while True:
        print("Change owed = ", end="")
        m = cs50.get_float()
        if m > 0:
            break

    m = int(round(m * 100))

    q = 0
    while m >= 25:
        m = m - 25
        q += 1

    a = int(round(m)) % 25

    d = 0
    while a >= 10:
        a = a - 10
        d += 1

    b = int(round(a)) % 10

    n = 0
    while b >= 5:
        b = b - 5
        n += 1

    p = 0
    p = int(round(b)) % 5

    print("Minimum number of coins used is {}".format(q + d + n + p))

if __name__ == "__main__":
    main()
